lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -279,7 +284,7 @@ struct StructuralHasher {
       // Skip names and annotations, except in certain cases.
       // Names of ports are load bearing for classes, so we do hash those.
       bool isClassPortNames =
-          isa<ClassLike>(op) && name == constants.portNamesAttr;
+          dedupClasses && isa<ClassLike>(op) && name == constants.portNamesAttr;
```

Comment:
I'm wondering how/why we might visit a class operation here anyway?

Is there an idea we might have a class operation at some point that isn't top-level (or what the StructuralHasher is instantiated on)?  (I don't think so, but that's one argument for this...)

Otherwise, maybe the StructuralHasher changes aren't needed, just keep the part that doesn't bother hashing class operations.

Reply:
But I think you're actually right, this doesn't need to be disabled. If the classes themselves don't dedup, their class names will always produce distinct hashes and prevent anything involving that class from deduping. Very insightful :1st_place_medal: :100: I'll tweak this.